lightning/src/ln/interactivetxs.rs

```diff
@@ -201,10 +175,32 @@ struct NegotiationContext {
 	holder_is_initiator: bool,
 	received_tx_add_input_count: u16,
 	received_tx_add_output_count: u16,
+	/// The inputs
 	inputs: HashMap<SerialId, InteractiveTxInput>,
+	/// The output intended to be the new funding output.
+	/// When an output added to the same pubkey, it will be treated as the shared output.
+	/// The script pubkey is used to discriminate which output is the funding output.
+	/// The value provided here is just a hint, when the actual output is added,
+	/// its value will take over.
+	intended_new_funding_output: TxOut,
+	/// The holder's intended contribution from the shared funding output,
+	/// in the range 0 -- intended_new_funding_output.value.
+	/// The rest is the counterparty's contribution
+	/// (intended_new_funding_output.value - intended_local_contribution_satoshis)
+	/// When the funding output is added (recognized by its output pubkey), it will be marked
+	/// as shared, split between the peers according to this value.
+	/// (In the rare case of actual output value differing from intended, the intended local
+	/// contribution may be clipped.)
+	intended_local_contribution_satoshis: u64,
+	/// The actual new funding output, set only after the output has been actually added.
+	/// Note: this output is also included in `outputs`.
```

Comment:
some small nits
```suggestion
	/// The output intended to be the new funding output.
	/// When an output is added with the same `script_pubkey`, it will be treated as the shared output.
	/// The `script_pubkey` field is used to determine which output is the funding output.
	/// The `value` provided here is just a hint, when the actual output is added,
	/// its `value` will take over.
	intended_new_funding_output: TxOut,
	/// The holder's intended contribution to the shared funding output,
	/// in the range [0, intended_new_funding_output.value].
	/// The rest is the counterparty's contribution
	/// (intended_new_funding_output.value - intended_local_contribution_satoshis)
	/// When the funding output is added (recognized by its output `script_pubkey`), it will be marked
	/// as shared, and split between the peers according to this value.
	/// (In the rare case of actual output value differing from intended, the intended local
	/// contribution may be clipped.)
	intended_local_contribution_satoshis: u64,
	/// The actual new funding output, set only after the output has actually been added.
	/// NOTE: This output is also included in `outputs`.
```